temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowClientInternal.java

```diff
@@ -414,4 +427,10 @@ public static WorkflowExecution start(Functions.Proc workflow) {
       A6 arg6) {
     return execute(() -> workflow.apply(arg1, arg2, arg3, arg4, arg5, arg6));
   }
+
+  private static void checkThread() {
+    if (Thread.currentThread().getName().startsWith("workflow")) {
```

Comment:
This doesn't look like a good method to check the thread for being a workflow. What if some user code names it's own thread "workflow"?

Looking at currently existing code, the right way to do this is
```
DeterministicRunnerImpl.currentThreadInternalIfPresent().isPresent();
```

I would actually add a method like
`WorkflowInternal#isCurrentThreadWokflow` containing this exact code.

Reply:
I'm confused why the above guarantees that this is a wf thread as opposed to just any other internal thread. 
Though I agree that relying on the name of the thread is not very robust. This is where I encountered this way of identifying threads:
[ExceptionInSignalTest.java#L69](https://github.com/temporalio/sdk-java/blob/ce9c1dc5f9105f8631092364f4022407c3ae4ec3/temporal-sdk/src/test/java/io/temporal/workflow/signalTests/ExceptionInSignalTest.java#L69)